CardPayments/src/main/java/com/paypal/android/cardpayments/CardRequest.kt

```diff
@@ -1,17 +1,24 @@
 package com.paypal.android.cardpayments
 
+import android.os.Parcelable
 import com.paypal.android.cardpayments.threedsecure.SCA
+import kotlinx.parcelize.Parcelize
 
 /**
  * A card request to process a card payment
  * @property orderId The order ID to to process the payment
  * @property card   Card used for payment
  * @property returnUrl Url to return to app after SCA challenge finishes
  * @property sca Specify to always launch 3DS or only when required. Defaults to `SCA.SCA_WHEN_REQUIRED`.
+ * @property shouldVault Set this value to true if the payment method should be vaulted on success
+ * @property vaultCustomerId Set a customer id for vaulting payment methods when [shouldVault] is true
  */
+@Parcelize
 data class CardRequest @JvmOverloads constructor(
     val orderId: String,
     val card: Card,
     val returnUrl: String,
-    val sca: SCA = SCA.SCA_WHEN_REQUIRED
-)
+    val sca: SCA = SCA.SCA_WHEN_REQUIRED,
+    val shouldVault: Boolean = false,
+    val vaultCustomerId: String? = null
```

Comment:
I know we had initially discussed 2 approaches to vaulting:
1. add everything to the card object then do the same with other modules, so for example vaulting PayPal or Venmo would also require us to have `shouldVault` and `vaultCustomerId`
2. Have and explicit Vault object to hold these parameters (`shouldVault` and `vaultCustomerId`)

I am curious if as these objects grow it will be harder to maintain having these in the individual objects vs a shared `Vault` object that can be used across multiple modules. I'm curious what your thoughts since I know you're a lot more familiar with the underlying APIs than I am!

Reply:
Yeah I just dropped a link to Jose's document on Vault API in the team chat.

Reply:
It's up to us. We can either throw or allow it and no-op (I believe the latter is the current behavior).

Reply:
Yeah, personally I think it'd be weird to no-op if a merchant provides a customer ID but should vault as false and could see us getting inbounds around that. It sounds like the options we have are:
```swift
// option 1: provide a Vault object and default should vault to true
let cardRequest = CardRequest(
    ...
    vault: Vault() // we generate customer ID
    // OR
    vault: Vault(customerID: "my-customer-id" // merchant provided customer ID
)

// option 2: provide a Vault object and have merchants provide should vault
// should we throw if a merchant tries to pass false
let cardRequest = CardRequest(
    ...
    vault: Vault(shouldVault: true) // we generate customer ID
    // OR
    vault: Vault(shouldVault: true, customerID: "my-customer-id" // merchant provided customer ID
)

// option 3: have vault options on the request directly
let cardRequest = CardRequest(
    ...
    shouldVault: true // we generate a customer ID - is this explicit enough for merchants to know that we generate the ID
    // OR
    shouldVault: true, 
    customerID: "my-customer-id" // merchant provided customer ID - what do we do in this case if a merchant provides an ID but should vault as false
)
```
Wanted to type them out to grock what the different options look like. Personally I still lean towards a vault object since it is more explicit. Not having one makes it a bit harder to handle the what ifs and doesn't make it clear that merchants can pass should vault and have us generate a customer ID for them in the top level object.

Reply:
Yeah I agree, they all field weird to me. At PPCP stand David mention we can go with Option 3 for now and circle back once we get some merchant feedback.

Reply:
Ah, gotcha, I didn't realize that was the decision I thought we wanted to mock them out a bit more and discuss. I think this is fine for now then and we can revisit later.

Reply:
I also lean toward a vault object. Something like:

```
let cardRequest = CardRequest(
    ...
    useVault: Vault() // we generate customer ID
    // OR
    useVault: Vault(customerID: "my-customer-id") // merchant provided customer ID
)
```

Reply:
Yeah we made the decision to go option 3 the other day in stand. I'm down to bring it up again in the team channel, we should get David involved if we decide to switch it up.

Reply:
I think we should circle back on it with David - I can post in there

Reply:
If we already decided on 3 we can stick with that! I know sometimes PPCP can get stuck in a lot of back and forth.

Reply:
Sounds like option 1 is the route. I'm going to change it to that one.